settings.gradle

```diff
@@ -87,30 +87,30 @@ if (!flinkVersions.isEmpty()) {
 }
 
 if (flinkVersions.contains("1.12")) {
-  include ':iceberg-flink:flink-1.12'
-  include ':iceberg-flink:flink-runtime-1.12'
-  project(':iceberg-flink:flink-1.12').projectDir = file('flink/v1.12/flink')
-  project(':iceberg-flink:flink-1.12').name = 'iceberg-flink-1.12'
-  project(':iceberg-flink:flink-runtime-1.12').projectDir = file('flink/v1.12/flink-runtime')
-  project(':iceberg-flink:flink-runtime-1.12').name = 'iceberg-flink-runtime-1.12'
+  include ":iceberg-flink:flink-1.12_${scalaVersion}"
+  include ":iceberg-flink:flink-runtime-1.12_${scalaVersion}"
+  project(":iceberg-flink:flink-1.12_${scalaVersion}").projectDir = file('flink/v1.12/flink')
+  project(":iceberg-flink:flink-1.12_${scalaVersion}").name = "iceberg-flink-1.12_${scalaVersion}"
+  project(":iceberg-flink:flink-runtime-1.12_${scalaVersion}").projectDir = file('flink/v1.12/flink-runtime')
+  project(":iceberg-flink:flink-runtime-1.12_${scalaVersion}").name = "iceberg-flink-runtime-1.12_${scalaVersion}"
 }
 
 if (flinkVersions.contains("1.13")) {
-  include ':iceberg-flink:flink-1.13'
-  include ':iceberg-flink:flink-runtime-1.13'
-  project(':iceberg-flink:flink-1.13').projectDir = file('flink/v1.13/flink')
-  project(':iceberg-flink:flink-1.13').name = 'iceberg-flink-1.13'
-  project(':iceberg-flink:flink-runtime-1.13').projectDir = file('flink/v1.13/flink-runtime')
-  project(':iceberg-flink:flink-runtime-1.13').name = 'iceberg-flink-runtime-1.13'
+  include ":iceberg-flink:flink-1.13_${scalaVersion}"
+  include ":iceberg-flink:flink-runtime-1.13_${scalaVersion}"
+  project(":iceberg-flink:flink-1.13_${scalaVersion}").projectDir = file('flink/v1.13/flink')
+  project(":iceberg-flink:flink-1.13_${scalaVersion}").name = "iceberg-flink-1.13_${scalaVersion}"
+  project(":iceberg-flink:flink-runtime-1.13_${scalaVersion}").projectDir = file('flink/v1.13/flink-runtime')
+  project(":iceberg-flink:flink-runtime-1.13_${scalaVersion}").name = "iceberg-flink-runtime-1.13_${scalaVersion}"
 }
 
 if (flinkVersions.contains("1.14")) {
-  include ':iceberg-flink:flink-1.14'
-  include ':iceberg-flink:flink-runtime-1.14'
-  project(':iceberg-flink:flink-1.14').projectDir = file('flink/v1.14/flink')
-  project(':iceberg-flink:flink-1.14').name = 'iceberg-flink-1.14'
-  project(':iceberg-flink:flink-runtime-1.14').projectDir = file('flink/v1.14/flink-runtime')
-  project(':iceberg-flink:flink-runtime-1.14').name = 'iceberg-flink-runtime-1.14'
+  include ":iceberg-flink:flink-1.14_${scalaVersion}"
+  include ":iceberg-flink:flink-runtime-1.14_${scalaVersion}"
+  project(":iceberg-flink:flink-1.14_${scalaVersion}").projectDir = file('flink/v1.14/flink')
+  project(":iceberg-flink:flink-1.14_${scalaVersion}").name = "iceberg-flink-1.14_${scalaVersion}"
+  project(":iceberg-flink:flink-runtime-1.14_${scalaVersion}").projectDir = file('flink/v1.14/flink-runtime')
+  project(":iceberg-flink:flink-runtime-1.14_${scalaVersion}").name = "iceberg-flink-runtime-1.14_${scalaVersion}"
```

Comment:
Correct me if I’m wrong, but I think we need to be able to publish / build all combinations of Flink / Scala versions here. For example, when publishing a new release to maven we need to build all of the artifacts.

Is it possible to do that in a way that doesn’t require two build calls (one for each Scala version)? I’m not a highly experienced Scala with gradle user, so I might be mistaken or it might be necessary to use two build calls.

But it makes the release process more cumbersome and particular to have to call build multiple times, which is something we’ve been trying to make more streamlined (the build / release process).

I’m ok with iterating on things and would not block on this, but it is a concern for me.

Reply:
Looking at the Spark on Scala 2.13 support PR, it looks like `scalaVersion` flag with two build calls is what is used.

I’m not sure if the release scripts have been updated to support this yet though.

Reply:
> Is it possible to do that in a way that doesn’t require two build calls (one for each Scala version)?

As far as I know, the `gradle` command cannot deploy the projects with two different values for a given system property. I mean I think the two build calls to build two separate flink scala runtime jar is the thing we need to do.  I see we usually deploy the artifacts to maven repository by executing this [stage-binaries.sh](https://github.com/apache/iceberg/blob/master/dev/stage-binaries.sh). I think it's okay to add the two `gradle` build calls into this bash script. 

(In general, I think it's a good question)

Reply:
Yeah, we can't have multiple Scala versions in the build without strange errors. We'll have to update the binary release script to publish Scala 2.13 modules.